code/parachain/frame/crowdloan-rewards/src/lib.rs

```diff
@@ -350,6 +378,54 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		/// Patch existing rewards by deleting them.
+		/// This will cause inconsistencies if a claim has already been made. Errors can be avoided
+		/// by passing the expected amount of rewards claimable.
+		///
+		/// Security:
+		/// * Should be called within a transactional context, as this may return an error but still
+		///   change
+		/// storage.
+		pub(crate) fn do_delete(
+			deletions: &[(RemoteAccountOf<T>, BalanceOf<T>)],
+		) -> DispatchResult {
+			let mut total_rewards: T::Balance = TotalRewards::<T>::get();
+			let mut total_contributors: u32 = TotalContributors::<T>::get();
+
+			// Get each account from deletions and remove it from the storage
+			for (account, expected) in deletions {
+				Rewards::<T>::mutate_exists(account, |rewards| {
+					if let Some(rewards) = rewards {
+						ensure!(rewards.total == expected, Error::<T>::UnexpectedRewardAmount);
+						total_rewards -= rewards.total;
+						total_contributors -= 1;
+					}
+					*rewards = None;
+					Ok::<(), Error<T>>(())
+				})?;
+			}
+
+			TotalRewards::<T>::set(total_rewards);
+			TotalContributors::<T>::set(total_contributors);
+			Ok(())
+		}
+
+		/// Add new rewards to the pallet. Updates total_rewards and contributors accordingly.
+		pub fn do_add(
+			additions: Vec<(RemoteAccountOf<T>, RewardAmountOf<T>, VestingPeriodOf<T>)>,
+		) -> DispatchResult {
+			Self::in_uninitialized(|| Self::do_populate(additions))
+		}
+
+		/// Runs the provided function while setting the crate to uninitialized. Useful during
+		/// patching.
+		pub(crate) fn in_uninitialized<F: FnOnce() -> DispatchResult>(f: F) -> DispatchResult {
+			let start = VestingTimeStart::<T>::take();
+			let result = f();
```

Comment:
We may need this validation from `do_initialize`
```
let excess_funds = available_funds
				.checked_sub(&total_rewards)
				.ok_or(Error::<T>::RewardsNotFunded)?;

```
To avoid a case of not having enough funds to support the rewards

Reply:
That plays poorly with 0ed updated; will add a check to do_populate.